File: BE/BookShopOnline/BookShopOnline.Core/Entitites/Book.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookShopOnline.Core.Entitites
{
    /// <summary>
    /// Class BaseEntity
    /// </summary>
    /// CreatedBy: LQHUY(16/03/2024)
    public class Book : BaseEntity
    {
        public Guid BookId { get; set; }
        [Required(ErrorMessage ="Mã sách không được để trống.")]
        public string BookCode { get; set; }
        [Required(ErrorMessage = "Tên sách không được để trống.")]
        public string BookName { get; set; }

      
        [Range(0, int.MaxValue, ErrorMessage = "Giá sách nhập vào phải là số.")]
        public double? Price { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Giá sách nhập vào phải là số.")]

        public double? OriginalPrice { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Giá sách nhập vào phải là số.")]

        public int? Discount { get; set; }

        [Required(ErrorMessage = "Tác giả không được để trống")]
        public string Author { get; set; }

        [Required(ErrorMessage = "Số lượng nhập không được để trống")]
        [Range(0, int.MaxValue, ErrorMessage = "Số lượng nhập vào phải là số.")]
        public int? QuantityImported { get; set; }
        public string? Description { get; set; }
        public DateTime? PublicationDate { get; set; }
        public string? Size { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Tổng số trang nhập vào phải là số.")]
        public int? NumberOfPage { get; set; }
        public string ImagePath { get; set; }
        public double? Heavy { get; set; }
        public Guid CategoryId { get; set; }
        public Guid PublisherId { get; set; }
        public string? CategoryName { get; set; }
        public string? PublisherName { get; set; }
        public int? QuantityInStock { get; set; }
        public int? QuantitySold { get; set; }
        public string? Slug { get; set; }
        public string? BookSlug { get; set; }
        public string? CategorySlug { get; set; }
        public string? PublisherSlug { get; set; }

    }
}

File: FE/bookshop/src/js/data/book.js
const bookColumns = [
  {
    field: "BookCode",
    nameField: "Mã sách",
    class: "table-align-text",
    width: "112px",
  },
  {
    field: "Image",
    nameField: "Hình ảnh",
    class: "table-align-text",
    width: "120px",
    textAlign: "center",
  },
  {
    field: "BookName",
    nameField: "Tên sách",
    class: "table-align-text",
    width: "352px",
  },
  {
    field: "Author",
    nameField: "Tác giả",
    class: "table-align-text",
    width: "182px",
  },
  {
    field: "Price",
    nameField: "Giá",
    class: "table-align-text",
    width: "132px",
    textAlign: "right",
    type: "money",
  },
  {
    field: "QuantityInStock",
    nameField: "Số lượng tồn",
    class: "table-align-text",
    width: "132px",
    textAlign: "right",
  },
  {
    field: "QuantitySold",
    nameField: "Số lượng bán",
    class: "table-align-text",
    width: "132px",
    textAlign: "right",
  },
  {
    field: "NumberOfPage",
    nameField: "Tổng số trang",
    class: "table-align-text",
    width: "132px",
    textAlign: "right",

  },

  {
    field: "Size",
    nameField: "Kích cỡ",
    class: "table-align-text",
    width: "122px",
  },

  {
    field: "CategoryName",
    nameField: "Tên danh mục",
    class: "table-align-text",
    width: "132px",
  },
  {
    field: "PublisherName",
    nameField: "Tên NXB",
    class: "table-align-text",
    width: "182px",
  },
  {
    field: "PublicationDate",
    nameField: "Ngày xuất bản",
    class: "table-align-text",
    width: "132px",
    textAlign: "center",
    type: "datetime",
  },
];
export default bookColumns;
